src/containers/category-list/CategoryList.jsx

```diff
@@ -22,10 +22,22 @@ const CategoryList = ({ limit, gridStyles }) => {
   const categoryName = searchParams.get('categoryName')
 
   const onResponse = (res) => {
+    console.log('Response received:', res)
+
     if (res?.items) {
-      setCategories([...categories, ...res.items])
-    }
-    if (res?.count <= itemsPerPage * page) {
+      if (categoryName) {
+        const filteredCategories = res.items.filter((category) =>
+          category.name.toLowerCase().includes(categoryName.toLowerCase())
+        )
```

Comment:
Yaroslav, I tried to remove the jumps, but I’m not sure that I did everything correctly (I changed the names of the styles so that there was no conflict).   I corrected the block Results Not Found. I also removed the unnecessary check.